src/ir/module-utils.h

```diff
@@ -510,6 +513,12 @@ inline void collectHeapTypes(Module& wasm,
   for (auto& curr : wasm.events) {
     counts.note(curr->sig);
   }
+  for (auto& curr : wasm.tables) {
+    counts.maybeNote(curr->type);
+  }
+  for (auto& curr : wasm.elementSegments) {
+    counts.maybeNote(curr->type);
+  }
```

Comment:
@tlively is this enough for making sure table & elem types are counted correctly everywhere, or have I missed something?